File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let intersection=new Set();
    nums1=new Set(nums1)
    for(let i of nums2){
        if(nums1.has(i) && !(intersection.has(i))){
            intersection.add(i)
        }
    }
    intersection=[...intersection]
    return intersection;
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set <int> nums1Set(nums1.begin(),nums1.end());
        unordered_set <int> nums2Set(nums2.begin(),nums2.end());
        vector <int> res={};
        for(int i : nums1Set){
            if(nums2Set.count(i)){
                res.push_back(i);
            }
        }
        return res;
    }
};